zerver/views/video_calls.py

```diff
@@ -168,32 +169,24 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
     return json_success()
 
 
-def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def get_bigbluebutton_url(
+    request: HttpRequest, user_profile: UserProfile, meeting_name: str = REQ()
+) -> HttpResponse:
     # https://docs.bigbluebutton.org/dev/api.html#create for reference on the API calls
     # https://docs.bigbluebutton.org/dev/api.html#usage for reference for checksum
     id = "zulip-" + str(random.randint(100000000000, 999999999999))
-    password = b32encode(secrets.token_bytes(7))[:10].decode()
-    checksum = hashlib.sha256(
-        (
-            "create"
-            + "meetingID="
-            + id
-            + "&moderatorPW="
-            + password
-            + "&attendeePW="
-            + password
-            + "a"
-            + settings.BIG_BLUE_BUTTON_SECRET
-        ).encode()
-    ).hexdigest()
+    password = b32encode(secrets.token_bytes(7))[:20].decode()
+    signer = Signer()
     url = append_url_query_string(
         "/calls/bigbluebutton/join",
-        urlencode(
+        urlencode(  # type: ignore[type-var] # https://github.com/python/typeshed/issues/4234
             {
-                "meeting_id": id,
-                "password": password,
-                "checksum": checksum,
-            }
+                "meeting_id": signer.sign(id),
+                "name": signer.sign(meeting_name),
+                "password": signer.sign(password),
+            },
+            quote_via=quote,
```

Comment:
I don't think this is a secure way to sign a token.  Generally, you always want to sign the entire object, not sign individual components of it -- otherwise, someone can mix and match pieces of multiple requests to do whatever they want.

We should construct a single object that details the entire request (meeting ID, name, and password), encode that in a string, and then sign that.